src/commcare_cloud/fab/utils.py

```diff
@@ -177,30 +171,27 @@ def _show_no_auth_message():
             "    $ cp {project_root}/config.example.py {project_root}/config.py\n"
             "Then edit {project_root}/config.py"
         ).format(project_root=PROJECT_ROOT))
-        username = input('Github username (leave blank for no auth): ')
+        username = input('Github username (leave blank for no auth): ') or None
         password = getpass('Github password: ') if username else None
-        if not username:
-            _show_no_auth_message()
-        return Github(
-            login_or_token=username or None,
-            password=password,
-        )
+        return (username, password)
     else:
-        if GITHUB_APIKEY is None:
-            _show_no_auth_message()
-        return Github(
-            login_or_token=GITHUB_APIKEY,
-        )
+        return (GITHUB_APIKEY, None)
+
+
+@memoized
+def _get_github():
+    login_or_token, password = _get_github_credentials()
+    if not login_or_token:
+        print(magenta(
+            "Warning: Creation of release tags is disabled. "
+            "Provide Github auth details to enable release tags."
+        ))
+    return Github(login_or_token=login_or_token, password=password)
 
 
 @memoized
 def _github_auth_provided():
-    try:
-        from .config import GITHUB_APIKEY
-    except ImportError:
-        return True  # user is prompted for auth
-    else:
-        return GITHUB_APIKEY is not None
+    return bool(_get_github_credentials()[0])
```

Comment:
I wanted to use the `Github` object itself here, but couldn't find anything handy describing whether credentials were provided without reaching into its internals.